doc/rst/language/spec/generics.rst

```diff
@@ -473,50 +473,77 @@ A queried domain may not be modified via the name to which it is bound
 Function Visibility in Generic Functions
 ----------------------------------------
 
-When resolving function calls made within generic functions, there is an
-additional source of visible functions. Besides functions visible to the
-generic function’s point of declaration, visible functions are also
-taken from one of the call sites at which the generic function is
-instantiated for each particular instantiation. The specific call site
-chosen is arbitrary and it is referred to as the *point of
-instantiation*.
+When resolving a function call, as defined in :ref:`Function_Resolution`,
+there is an additional source of visible functions when the call is
+nested within a generic function. Those are the functions visible
+from the call site that the enclosing generic function is invoked from.
+This call site is referred to as the *point of instantiation*.
+If there are multiple enclosing generic functions or the call is nested
+within a concrete function that is, in turn, nested in generic function(s),
+the point of instantiation is the call site of the innermost generic function.
+
+If no candidate functions are found during the initial steps of identifying
+visible and candidate functions, function resolution performs these two steps
+at the point of instantiation. If still no candidate functions are found
+and the point of instantiation is nested within generic function(s),
+the point of instantiation of the innermost enclosing generic function
+is consulted. This process is repeated until the point of instantiation
+for which candidate function(s) are found. If points of instantiation
```

Comment:
> This process is repeated until the point of instantiation for which candidate function(s) are found

I'm not following this part, maybe there is a way to rephrase it?

Maybe

> This search for candidate function continues across points of instantiations until candidate functions are found or the point-of-instantiations are exhausted.

I wonder - are the point-of-instantiations exhausted when we are in a concrete function? Would that help the exposition?